default/AI/DiplomaticCorp.py

```diff
@@ -0,0 +1,78 @@
+"""Handle diplomatic messages and response determination."""
+
+import random
+
+import freeOrionAIInterface as fo  # pylint: disable=import-error
+import FreeOrionAI as foAI
+import AIstate
+from freeorion_tools import UserString, UserStringList, chat_on_error, print_error
+
+
+@chat_on_error
+def handle_diplomatic_message(message):  # pylint: disable=invalid-name
+    """Handle a diplomatic message update from the server,
+    such as if another player declares war, accepts peace, or cancels a proposed peace treaty.
+    :param message:
+    :return:
+    """
+    print "Received diplomatic %s message from empire %s to empire %s" % (message.type, message.sender, message.recipient)
+    print "my empire id: %s" % fo.empireID()
+    if message.recipient != fo.empireID():
+        return
+    reply_sender = message.recipient
+    reply_recipient = message.sender
+    if message.type == fo.diplomaticMessageType.peaceProposal:
+        foAI.foAIstate.log_peace_request(message.sender, message.recipient)
+        proposal_sender_player = fo.empirePlayerID(message.sender)
+        suffix = "MILD" if foAI.foAIstate.aggression <= fo.aggression.typical else "HARSH"
+        possible_acknowledgments = UserStringList("AI_PEACE_PROPOSAL_ACKNOWLEDGEMENTS_" + suffix + "_LIST")
+        acknowledgement = random.choice(possible_acknowledgments)
+        print "Acknowledging proposal with initial message (from %d choices): '%s'" % \
+              (len(possible_acknowledgments), acknowledgement)
+        fo.sendChatMessage(proposal_sender_player, acknowledgement)
+        attitude = evaluate_diplomatic_attitude(message.sender)
+        if attitude > 0:
+            reply_text = random.choice(UserStringList("AI_PEACE_PROPOSAL_RESPONSES_YES_" + suffix + "_LIST"))
+            diplo_reply = fo.diplomaticMessage(reply_sender, reply_recipient, fo.diplomaticMessageType.acceptProposal)
+        else:
+            reply_text = random.choice(UserStringList("AI_PEACE_PROPOSAL_RESPONSES_NO_" + suffix + "_LIST"))
+            diplo_reply = None
+        fo.sendChatMessage(proposal_sender_player, reply_text)
```

Comment:
I think it is a reasonable call either way. Once I have the AIs counter declaring war, I initially want it to irritate them slightly to be asked for peace, even if they agree.  Obviously, a more complex treatment wouldn't have to have that.